server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -174,12 +174,14 @@ public Function<byte[], Number> pointReaderIfPossible() {
         return null;
     }
 
-    /** Returns true if the field is aggregatable.
-     *
+    /**
+     * Returns true if the field is aggregatable.
+     * @param isSyntheticSource is the {@code _source} stored in synthetic
+     *                          {@code _source}
      */
-    public boolean isAggregatable() {
+    public boolean isAggregatable(boolean isSyntheticSource) {
```

Comment:
So this feels wrong as well, because whether or not the source is synthetic is fixed by the mappings, so we really shouldn't need to pass this in as a boolean because it will always be the same for a given MappedFieldType instance.  But I think the source of the wrongness is actually this default method implementation, which creates a field data builder and then checks for an exception.  This might be expensive! And something might not throw an exception at construction time but at segment load time!

What I think we should do is remove this default impl entirely, and let the different field types provide their own impls.  I'm happy to do this, I can start on something tomorrow.